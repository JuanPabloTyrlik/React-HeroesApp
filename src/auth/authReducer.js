import { TYPES } from '../types/types';

const initialState = {
    logged: false,
};

export const authReducer = (state = initialState, { type, payload }) => {
    switch (type) {
        case TYPES.LOGIN:
            return { ...payload, logged: true };
        case TYPES.LOGOUT:
            return { logged: false };
        default:
            return state;
    }
};
