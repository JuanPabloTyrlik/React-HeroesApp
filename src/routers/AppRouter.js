import React from 'react';
import { useContext } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRouter } from './DashboardRouter';
import { PrivateRoute } from './PrivateRoute';

export const AppRouter = () => {
    const {
        user: { logged },
    } = useContext(AuthContext);

    return (
        <Router>
            <div>
                <Switch>
                    <Route exact path="/login" component={LoginScreen} />
                    <PrivateRoute
                        path="/"
                        component={DashboardRouter}
                        isAuthenticated={logged}
                    />
                </Switch>
            </div>
        </Router>
    );
};
