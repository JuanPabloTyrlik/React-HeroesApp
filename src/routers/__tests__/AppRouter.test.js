import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('Tests on AppRouter', () => {
    test('should render login page if unauthorized', () => {
        const contextValue = {
            user: {
                logged: false,
            },
            dispatch: jest.fn(),
        };
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );
        expect(wrapper.find('h1').text()).toMatch('Login Screen');
        expect(wrapper.find('input').prop('placeholder')).toMatch('Username');
        expect(wrapper.find('button').text()).toMatch('Login');
    });
    test('should render Marvel Screen if authorized', () => {
        const contextValue = {
            user: {
                name: 'Juan Pablo',
                logged: true,
            },
            dispatch: jest.fn(),
        };
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper.find('.navbar').exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});
