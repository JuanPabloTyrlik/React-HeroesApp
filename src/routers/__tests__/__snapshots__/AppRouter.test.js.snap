// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on AppRouter should render Marvel Screen if authorized 1`] = `
<div>
  Array [
    <nav
      className="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Comics
      </a>
      <div
        className="navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-item nav-link active"
            href="/marvel"
            onClick={[Function]}
            style={Object {}}
          >
            Marvel
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </div>
      </div>
      <div
        className="navbar-collapse collapse w-100 order-3 dual-collapse2"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <form
            className="form-inline my-2 my-lg-0"
            onSubmit={[Function]}
          >
            <input
              aria-label="Search"
              className="form-control mr-sm-2"
              onChange={[Function]}
              placeholder="Search Hero"
              type="search"
              value=""
            />
          </form>
          <span
            className="navbar-text text-success"
          >
            Juan Pablo
          </span>
          <button
            className="nav-item nav-link btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>,
    <div
      className="container mt-2"
    >
      <div>
        <h1>
          Marvel
        </h1>
        <hr />
        <div
          className="card-columns animate__animated animate__fadeIn"
        >
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Spider Man"
                  className="card-img"
                  src="./assets/heroes/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Captain America"
                  className="card-img"
                  src="./assets/heroes/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Captain America Comics #1
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Iron Man"
                  className="card-img"
                  src="./assets/heroes/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #39
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Thor"
                  className="card-img"
                  src="./assets/heroes/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Journey into Myster #83
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Hulk"
                  className="card-img"
                  src="./assets/heroes/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Wolverine"
                  className="card-img"
                  src="./assets/heroes/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Daredevil"
                  className="card-img"
                  src="./assets/heroes/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Daredevil #1
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Hawkeye"
                  className="card-img"
                  src="./assets/heroes/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #57
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Cyclops"
                  className="card-img"
                  src="./assets/heroes/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      X-Men #1
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img"
                  src="./assets/heroes/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Fantastic Four #48
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card ms-3"
            style={
              Object {
                "cursor": "pointer",
                "maxWidth": 540,
              }
            }
          >
            <div
              className="row no-gutters"
              onClick={[Function]}
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="The Punisher"
                  className="card-img"
                  src="./assets/heroes/marvel-punisher.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    The Punisher
                  </h5>
                  <p
                    className="card-text"
                  >
                    Frank Castle
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Amazing Spider-Man # 129
                    </small>
                  </p>
                  <span
                    className="link"
                  >
                    Más...
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>,
  ]
</div>
`;
